### DISCUSSION TITLE
Create a CAS Provider

### DISCUSSION THREAD

**Discussion by nikero41** [nextauthjs/next-auth#7909] [Ideas]
### Description 📓

I am trying to authenticate using an external CAS server. It would be nice if there was a CAS Provider to help with this.
e.x. [node-cas-client](https://www.npmjs.com/package/node-cas-client)

### How to reproduce ☕️

Create a CAS server and a minimal Next JS app with next-auth and create a provider to authenticate to the CAS sever

### Contributing 🙌🏽

Yes, I am willing to help implement this feature in a PR

**Comment by balazsorban44** [comment]
As I understand, CAS is a different spec than OAuth, what we currently support. This makes it non-trivial to add support quickly, so I'm moving this to a discussion for now.

**Reply by ArturoMendes** [reply]
I understand it is not easy to implement CAS authentication but it is a shame to have to rely on Java implementations of CAS such a Apereo or Jasper when Next was just the perfect opportunity to have it implemented (I might be too ignorant to see any pitfalls). Just saying that it would be great to have that possibility.

**Comment by yalegria** [comment]
Hello, I am also looking for this integration, I thought Nextjs-auth.js already had it. As I understand the nextjs app will be the client to authenticate against a remote CAS server. As a probe of concept I created a simple nodejs app and use a package called http-cas-client and I was able to setup the integration easily. Why this is so complicated on Nextjs?

**Reply by paulm17** [reply]
Did you put that project on github?

**Comment by microfsconsulting** [comment]
Hello,
I need it to my enterprise :-)
I would like to use nextjs at work for internal application.

Thanks in advance.

**Comment by mkhawam** [comment]
Has this been worked on or discussed further. Many of the universities use cas as their main authorization method. Added cas authorization to next js would help a lot to get next js adapted at the universities.

**Reply by mkhawam** [reply]
For those who might be looking for next cas authentication, I found this [repo](https://github.com/uhawaii-system-its-ti-iam/next-cas-client) maintained by University of Hawaii. It provides a custom endpoint for cas transactions.
